src/scipp/ndimage/__init__.py

```diff
@@ -0,0 +1,247 @@
+    if not isinstance(index, Variable):
+        return index
+    coord = x.coords[dim]
+    if not islinspace(coord, dim).value:
+        raise CoordError(
+            f"Data points not regularly spaced along {dim}. To ignore this, "
+            f"provide a plain value (convertible to {dtype.__name__}) instead of a "
+            "scalar variable. Note that this will correspond to plain positional "
+            "indices/offsets.")
+    pos = (len(coord) - 1) * (index.to(unit=coord.unit) / (coord[-1] - coord[0])).value
+    return dtype(pos)
+
+
+def _positional_index(x: Union[Variable, DataArray], index, name=None, dtype=int):
+    if not isinstance(index, dict):
+        return [_delta_to_positional(x, dim, index, dtype=dtype) for dim in x.dims]
+    if set(index) != set(x.dims):
+        raise ValueError(f"Data has dims={x.dims} but input argument '{name}' provides "
+                         f"values for {tuple(index)}")
+    return [_delta_to_positional(x, dim, index[dim], dtype=dtype) for dim in x.dims]
+
+
+@_ndfilter
+def gaussian_filter(x: Union[Variable, DataArray],
+                    /,
+                    *,
+                    sigma,
+                    order=0,
+                    **kwargs) -> Union[Variable, DataArray]:
+    """
+    Multidimensional Gaussian filter.
+
+    This is a wrapper around :py:func:`scipy.ndimage.gaussian_filter`. See there for
+    full argument description. There are two key differences:
+
+    - This wrapper uses explicit dimension labels in the ``sigma`` and ``order``
+      arguments. For example, instead of ``sigma=[4, 6]`` use
+      ``sigma={{'time':4, 'space':6}}``
+      (with appropriate dimension labels for the data).
+    - Coordinate values can be used (and should be preferred) for ``sigma``. For
+      example, instead of ``sigma=[4, 6]`` use
+      ``sigma={{'time':sc.scalar(5.0, unit='ms'), 'space':sc.scalar(1.2, unit='mm')}}``.
+      In this case it is required that the corresponding coordinates exist and form a
+      "linspace", i.e., are evenly spaced.
+
+    Warning
+    -------
+    When ``sigma`` is an integer or an mapping to integers coordinate values are
+    ignored. That is, the filter is applied even if the data points are not evenly
+    spaced. The resulting filtered data may thus have no meaningful interpretation.
+
+    Parameters
+    ----------
+    x:
+        Input variable or data array.
+    sigma:
+        Standard deviation for Gaussian kernel. The standard deviations of the Gaussian
+        filter are given as a mapping from dimension labels to numbers, or as a single
+        number, in which case it is equal for all axes.
+    order:
+        The order of the filter along each dimension, given as mapping from dimension
+        labels to integers, or as a single integer. An order of 0 corresponds to
+        convolution with a Gaussian kernel. A positive order corresponds to convolution
+        with that derivative of a Gaussian.
+
```

Comment:
Note that you need to put a `Returns` section to properly show the return type in the docs.